supplementary/alica_tracing/src/tracing/Trace.cpp

```diff
@@ -18,19 +30,20 @@ Trace::~Trace()
     _rawTrace->Finish();
 }
 
-void Trace::setTag(const std::string& key, const std::string& value)
+void Trace::setTag(const std::string& key, const TraceValue& value)
 {
-    _rawTrace->SetTag(key, value);
+    _rawTrace->SetTag(key, prepareRawTraceValue(extractVariant(value)));
```

Comment:
I think we should take the parameter `value` by value & then std::move here. The way it is right now will always make copies I think i.e.
```
void Trace::setTag(const std::string& key, TraceValue value) {  _rawTrace->SetTag(key, prepareRawTraceValue(extractVariant(std::move(value))); }
```

Reply:
Changed to use string views, this might minimize the required copying. Although does look bit fragile.